core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -1088,7 +1089,10 @@ private static SslContext validateSslContext(SslContext sslContext, boolean tlsA
             clientEngine.wrap(appBuf, packetBuf);
             appBuf.clear();
             packetBuf.flip();
-            serverEngine.unwrap(packetBuf, appBuf);
+            final SSLEngineResult result = serverEngine.unwrap(packetBuf, appBuf);
+            if (result.bytesConsumed() == 0) {
+                throw new IllegalStateException("failed to validate SSL/TLS configuration: " + sslContext);
+            }
```

Comment:
`ServerTlsValidationTest.testPkcs12KeyStoreWithNullPassword()` failure is caused by https://github.com/netty/netty/pull/10920 applied to Netty [4.1.57](https://netty.io/news/2021/01/12/4-1-57-Final.html).
An SSLException is not raised anymore on an invalid TLS configuration.

Reply:
An SSLException raised while reading encrypted data is handled by `handleUnwrapException(...)`
https://github.com/netty/netty/blob/3353bc115217ccb292c95680da063212ae94d533/handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java#L1217-L1222
and returned an `SSLEngineResult` with `OK` status.
https://github.com/netty/netty/blob/3353bc115217ccb292c95680da063212ae94d533/handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java#L1335-L1337
So `result.bytesConsumed()` is used for checking correctness.